extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/ConfigStaticInitValues.java

```diff
@@ -0,0 +1,86 @@
+package io.quarkus.arc.runtime;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import jakarta.annotation.Priority;
+import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.inject.spi.Annotated;
+import jakarta.enterprise.inject.spi.AnnotatedConstructor;
+import jakarta.enterprise.inject.spi.AnnotatedField;
+import jakarta.enterprise.inject.spi.AnnotatedParameter;
+import jakarta.enterprise.inject.spi.InjectionPoint;
+import jakarta.inject.Singleton;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.config.ConfigValue;
+
+import io.quarkus.runtime.StartupEvent;
+import io.smallrye.config.SmallRyeConfig;
+
+@Singleton
+public class ConfigStaticInitValues {
+
+    private final List<InjectedValue> injectedValues = new CopyOnWriteArrayList<>();
```

Comment:
During static init, this list is only written to, and then it's read once during startup, so COW list doesn't seem like the best choice. A simple synchronized list would probably work better.